### ISSUE TITLE
Embeddable object fails to load date

### ISSUE THREAD

**Issue by AzariasB** [mikro-orm/mikro-orm#5216]
### Describe the bug

When declaring an embeddable entity that contains a date property, using it as an object and trying to load it, the value is null instead of having the date.
The issue does not happen when using in-memory sqlite database

### Reproduction

Inspired from the embeded test for postgres
```ts
import { Embeddable, Embedded, Entity, MikroORM, PrimaryKey, Property } from '@mikro-orm/core';
import { PostgreSqlDriver } from '@mikro-orm/postgresql';

@Embeddable()
class Profile {

  @Property()
  username: string;

  @Property({ nullable: false })
  createdAt: Date = new Date();

  constructor(username: string) {
    this.username = username;
    this.createdAt = new Date();
  }

}

@Entity()
class User {

  @PrimaryKey()
  id!: number;

  @Embedded(() => Profile, { object: true })
  profile1!: Profile;

}

describe('embeddables date should load', () => {

  let orm: MikroORM<PostgreSqlDriver>;

  beforeAll(async () => {
    orm = await MikroORM.init({
      entities: [User],
      driver: PostgreSqlDriver,
      dbName: `mikro_orm_embeddables_date_bug`,
    });
    await orm.schema.refreshDatabase();
  });

  afterAll(async () => {
    await orm.close(true);
  });

  async function createUser() {
    const user1 = new User();
    user1.profile1 = new Profile('u2');

    await orm.em.persistAndFlush(user1);
    orm.em.clear();

    return user1;
  }

  test('persist and load', async () => {
    const  user = await createUser();
    const u = await orm.em.findOneOrFail(User, user.id);
    expect(u.profile1.createdAt).toEqual(user.profile1.createdAt);
  });

});
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.1.0

### Node.js version

21.6.1

### Operating system

ubuntu 22.04

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Interesting, the problem seems to be parsing via `postgres-date` which returns `null` for ISO date strings. Apparently, it does not like the `T` that connects date and time, `parseDate(value.replace('T', ' ')` seems to help.

**Comment by B4nan** [comment]
That library was already used under the hood by `node-postgres` (via [pg-types package](https://github.com/brianc/node-pg-types/blob/master/package.json#L43)), so that worked the same since inception, it was always used for date parsing. I don't think it's unmaintained, there is just not much to maintain on such a small and low level library. 

Now it's used explicitly also for converting dates inside embedded properties, so inside JSON properties.

https://github.com/mikro-orm/mikro-orm/commit/e774d4092eb13d81772701f7d8e35ada55fb4d45?w=1

**Comment by AzariasB** [comment]
https://github.com/bendrucker/postgres-date/issues/38 created, will see how it goes :D

**Comment by B4nan** [comment]
I fixed it here so we are not blocked by upstream, we can revert that once its fixed in postgres-date.

**Comment by AzariasB** [comment]
Thank you for the fast fix!

**Comment by AzariasB** [comment]
Welp, [out of scope for postgres-date](https://github.com/bendrucker/postgres-date/issues/38#issuecomment-1933256880), so it looks like the fix is here to stay.
